### PR TITLE
[WIP] integrate regionskate

### PR DESCRIPTION
**Pull request by ljwolf**
this is a work in progress tracking PR to integrate the skater module. 

Don't close. I'll ping when ready. 

Thanks!

ljw

### CODE REVIEW

**Comment by ljwolf** [comment]
OK, this rolls a bunch of stuff plus the skater integration in regionskate together now. 

@yogabonito I will enable travis rn, but otherwise I'll merge this myself

**Comment by ljwolf** [comment]
seems like tests will fail until [this](https://github.com/cython/cython/issues/1715) is resolved?

**Comment by ljwolf** [comment]
This is only on hold for the reason identified above in python 3.6 for some reason. 

In python 3.5, #7 occurs. Not sure what to do here except bone up on pytest.

**Comment by ljwolf** [comment]
Nearly there. 

`pytest` stlil considers the warning raised in `Spanning_Tree(n_clusters=np.inf)` an error, so I'm trying modifying the `pytest.ini`. 

Second, there still are tons of substantive failures in non-spanning tree code in 3.6...

**Comment by ljwolf** [comment]
Errors are now in the substantive code for the `networkx` and `scipy.sparse` fit methods only for 3.6. Maybe there was an `nx` ordering change? I'm at a loss on this atm.

**Comment by ljwolf** [comment]
I fixed the `networkx`-related errors back in `libpysal` and now this passes.

**Comment by ljwolf** [comment]
I fixed the `networkx`-related errors back in `libpysal` and now this passes.

File: pytest.ini [hunk 1]
```diff
@@ -1,2 +1,4 @@
 [pytest]
-addopts = --doctest-modules
\ No newline at end of file
+addopts = --doctest-modules
+filterwarnings =
```
Comment (weikang9009) [inline pytest.ini]: This configuration will suppress `pytest` with  `travis` to capture or show `optimizewarning`.  It is very weird that when testing locally with `pytest`, it won't fail but to give a warning, while when building with `travis`, `pytest` will always fail and exit with code 1.
image

**Pull request merged**